### ISSUE TITLE
BOP world type does not generate modded biomes like in 1.12.2

### ISSUE THREAD

**Issue by Lykrast** [Glitchfiend/BiomesOPlenty#1428]
### What's the issue you encountered?

BOP world type does not generate modded biomes like it did in 1.12.2.

### How can the issue be reproduced?

For 1.12.2:
* BiomesOPlenty-1.12.2-7.0.1.2439
* [Traverse-1.12.2-1.6.0-69](https://www.curseforge.com/minecraft/mc-mods/traverse-legacy)
* [Redwoods-1.12-1.2.1](https://www.curseforge.com/minecraft/mc-mods/redwoods)
* [Nature's Compass](https://www.curseforge.com/minecraft/mc-mods/natures-compass) to find biomes

Generated a BOP world, and the compass could find several of the biomes from those other mods (like Mini Jungle or Redwood Forest (the one with the Redwoods trees, it could also find one with the BOP ones)), which all generated alongside the BOP biomes.

For 1.14.4:
* BiomesOPlenty-1.14.4-9.0.0.195
* [Voyage-1.0.0](https://www.curseforge.com/minecraft/mc-mods/voyage)
* [LocateBiome](https://www.curseforge.com/minecraft/mc-mods/locatebiome) to find biomes (there's a fabric and a forge version)

Generated a BOP world again. The `locatebiome` command found BOP biomes fine but could not find any single Voyage biome (the ones whose registry name starts with `voyage:`) within the 2 minutes timeout.

**Comment by Forstride** [comment]
I don't have an actual reason as to why this isn't working (@Adubbz handles most of the advanced coding stuff), but honestly, it's probably for the better.  With our climate system, it'd be ideal if modders adding overworld biomes used our API to fine-tune where their biomes appear.  You can just check if BOP is installed and then add your biomes to the appropriate climate(s).

You can check https://github.com/Glitchfiend/BiomesOPlenty/blob/BOP-1.14.x-9.x.x/src/main/java/biomesoplenty/api/enums/BOPClimates.java for an example of what vanilla biomes are registered to which climates (And individual BOP biome classes for those).

**Comment by Lykrast** [comment]
That's a bit of a shame (would have liked a best guess default approach) but that API seems clear enough to use.

The README is missing the maven info to add to workspaces though so where can I find that?

**Comment by Forstride** [comment]
Uhh, I think you could just use CurseForge's built-in one or whatever.  Although we use Forge's for our builds: http://files.minecraftforge.net/maven/com/github/glitchfiend/biomesoplenty/BiomesOPlenty/index_1.14.4.html

(Not sure if any more info is needed, cause I don't handle this kind of stuff)

**Comment by Lemonszz** [comment]
+1 to automatically doing this again.

Asking every mod that adds a biome to generation to use your API seems a bit presumptuous to me, especially with some mods only adding a few (that could even be vital to progression).

Taking a best guess approach using the biome dictionary/temperature/category/precipitation/downfall data would be appreciated.

**Comment by Forstride** [comment]
Well to be fair, BOP really isn't meant to be used with other biome mods, especially since there's often a lot of overlap.

But I'll re-open it for the time being.  No idea when it'll be fixed though, because we have a bunch of other stuff we have to deal with still, and it's really not much of a priority right now.  And when I say we, I mostly mean Adubbz since he handles all the advanced stuff, but he's been pretty busy with IRL stuff and other hobbies of his.

**Comment by Syndaryl** [comment]
I'm happy with overlap, and if I don't want it, generally I can disable unwanted redundant biomes myself. As an example, three different varieties of redwood forest may seem redundant, but each likely has its own tree types and shapes, its own mobs (possibly including custom mobs with important drops), its own special ore-type resources, or its own decorative plants. Being able to dip another mod where desired to add a biome varient or exotic special biome lets me choose to selectively expand without nagging y'all for features you aren't interested in implementing.

Basically, I'm saying "thank you!" For reopening this ticket. I understand it being a low priority, and currently 1.14 is still not as diverse a mod landscape as 1.12 so it's not yet as big an issue anyways. It's good to know that in the future when there are more biome mods kicking around, this feature might be implemented.
